c++/src/kj/compat/http.c++

```diff
@@ -3435,14 +3441,16 @@ private:
 
     currentlySending = true;
 
-    KJ_IF_SOME(p, sendingPong) {
-      // We recently sent a pong, make sure it's finished before proceeding.
-      auto promise = p.then([this, opcode, message]() {
-        currentlySending = false;
-        return sendImpl(opcode, message);
-      });
-      sendingPong = kj::none;
-      return promise;
+    while (sendingControlMessage != kj::none) {
+      KJ_IF_SOME(p, sendingControlMessage) {
```

Comment:
Nitpick, feel free to ignore:

There's some part of my brain that really doesn't like that this is checking the same thing the previous line was checking.

I think I would normally write this as `for (;;) { KJ_IF_SOME(...) { ... } else { break; } }`. Which is not pretty either but avoids the duplicate check.

That said I wonder if we should create a `KJ_WHILE_SOME`. The pattern has come up a few times.

Reply:
Done.

I'm in favor of `KJ_WHILE_SOME`. We've got a branching construct for dealing with `Maybe`; let's also have a loop.